bin/benchmarks-to-datadog

```diff
@@ -0,0 +1,173 @@
+
+    abort_with_help(parser, "CSV file is missing") if csv_file.to_s.empty?
+    abort_with_help(parser, "API key is missing") if api_key.to_s.empty?
+    abort_with_help(parser, "App key is missing") if app_key.to_s.empty?
+
+    new(api_key: api_key, app_key: app_key).ingest_csv(csv_file)
+  end
+
+  def self.abort_with_help(parser, message)
+    $logger.warn message
+    Optimist::with_standard_exception_handling(parser) { raise Optimist::HelpNeeded }
+  end
+
+  def ingest_csv(file_path)
+    total_rows = 0
+
+    CSV.foreach(file_path, headers: true) do |row|
+      next unless valid_benchmark_row?(row)
+
+      total_rows += 1
+      benchmark_name = row['benchmark']
+      tags = build_tags(row)
+      timestamp = parse_timestamp(row['timestamp'])
+
+      $logger.info "Processing row #{total_rows}: #{benchmark_name} at #{row['timestamp']}"
+
+      process_cpu_metrics(row, tags, timestamp)
+      process_run_metrics(row, tags, timestamp)
+      process_total_metrics(row, tags, timestamp)
+
+      $logger.info "Completed processing #{benchmark_name}"
+      $logger.info '-' * 60
+
+      sleep(0.1)
+    end
+
+    $logger.info "Ingestion Summary:"
+    $logger.info "Total rows processed: #{total_rows}"
+  end
+
+  def valid_benchmark_row?(row)
+    row['benchmark'] && !row['benchmark'].strip.empty?
+  end
+
+  def parse_timestamp(timestamp_str)
+    Time.strptime(timestamp_str, "%a %b %d %H:%M:%S GMT %Y").to_i
+  rescue ArgumentError
+    Time.now.to_i
+  end
+
+  def build_tags(row)
+    tags = ["env:platforms-testing", "benchmark:#{row['benchmark']}"]
+    tags << "cpu_intensive:true" if row['cpu'] == 'true'
+    tags << "memory_intensive:true" if row['memory'] == 'true'
+    tags << "rendering_enabled:true" if row['rendering'] == 'true'
+    tags
+  end
+
+  def send_metric(name, value, tags, type = DatadogAPIClient::V2::MetricIntakeType::GAUGE, timestamp = Time.now.to_i)
+    metric_series = DatadogAPIClient::V2::MetricSeries.new(
+      metric: name,
+      type: type,
+      points: [
+        DatadogAPIClient::V2::MetricPoint.new(timestamp: timestamp, value: value)
+      ],
+      tags: tags,
+      resources: [
+        DatadogAPIClient::V2::MetricResource.new(name: @host, type: "host")
+      ]
+    )
+
+    payload = DatadogAPIClient::V2::MetricPayload.new(series: [metric_series])
+
+    begin
+      @api.submit_metrics(payload)
+    rescue DatadogAPIClient::APIError => e
+      $logger.error "Failed to send metric #{name}: #{e.message}"
+      $logger.error "HTTP status code: #{e.code}"
+      $logger.error "Response body: #{e.response_body}"
+    end
+  end
+
+  def process_cpu_metrics(row, tags, timestamp)
+    cpu_values = (1..5).map { |i| row["cpuUse.#{i}"]&.to_f }.compact
+    return if cpu_values.empty?
+
+    avg_cpu = cpu_values.sum / cpu_values.size
+    send_metric('benchmark.cpu_usage_percent', avg_cpu, tags, timestamp)
+  end
+
+  def process_run_metrics(row, base_tags, timestamp)
+    (1..5).each do |i|
```

Comment:
I suspect that giving `5` a name like `RUN_COUNT` (or similar) will help future travelers if the value ever needs to change.